### ISSUE TITLE
Does partial updating requires to load all data?

### ISSUE THREAD

**Issue by tleytens** [algolia/gatsby-plugin-algolia#121]
Hi

When testing the plugin I came to the following conclusion...

1) after an update in Kentico Kontent the Gatsby build is triggered.
2) Graphql loads all available content from Kentico Kontent.
3) Gatsby performs a search to retrieve the matchFields of all available content in Algolia.
4) Gatsby does local matching to see what nodes needs to be updated/deleted/created based on comparing both arrays.

My question... concerns.... Does this scale when you have a real high number of content? 
Or am I missing something in my current implementation / insights?

Kind regards

**Comment by Haroenv** [comment]
No, when the objects would no longer fit in memory, that could be an issue, in which case it can be more interesting to sync the Algolia index directly to algolia without involving the Gatsby plugin
